### PR TITLE
Add warnings for unexpected NodeTypes

### PR DESCRIPTION
**Pull request by nakabonne**
I solved the issue below.
Could someone please review?

https://github.com/istio/istio/issues/9249

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 1]
```diff
@@ -45,6 +46,8 @@ func (configgen *ConfigGeneratorImpl) BuildHTTPRoutes(env *model.Environment, no
 		return configgen.buildSidecarOutboundHTTPRouteConfig(env, node, push, proxyInstances, services, routeName), nil
 	case model.Router, model.Ingress:
 		return configgen.buildGatewayHTTPRouteConfig(env, node, push, proxyInstances, services, routeName)
+	default:
+		log.Warnf("BuildListeners Unexpected NodeType %s", node.Type)
```
Comment (supereagle) [inline pilot/pkg/networking/core/v1alpha3/httproute.go]: BuildListeners?

Comment (nakabonne) [reply pilot/pkg/networking/core/v1alpha3/httproute.go]: I'll fix it now.

**Comment by nakabonne** [comment]
@supereagle 
Thank you for review.
I fixed it, so please check it.

**Comment by supereagle** [comment]
LGTM

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 2]
```diff
@@ -45,6 +46,8 @@ func (configgen *ConfigGeneratorImpl) BuildHTTPRoutes(env *model.Environment, no
 		return configgen.buildSidecarOutboundHTTPRouteConfig(env, node, push, proxyInstances, services, routeName), nil
 	case model.Router, model.Ingress:
 		return configgen.buildGatewayHTTPRouteConfig(env, node, push, proxyInstances, services, routeName)
+	default:
```
Comment (sdake) [inline pilot/pkg/networking/core/v1alpha3/httproute.go]: Is this condition possible?  I find it unlikely that bad input would be passed in...  If there are other NodeTypes that can be decoded, seems like there should be logic to handle them.

Comment (nakabonne) [reply pilot/pkg/networking/core/v1alpha3/httproute.go]: Thank you for your review.
I also agree it's very unlikely.
However, I think that this method should not know whether it's handled outside.
So I added default handling.

**Comment by nakabonne** [comment]
Is not it necessary?

**Pull request closed without merging**